src/lib/middleware/assetMiddleware.ts

```diff
@@ -0,0 +1,40 @@
+import path from "path"
+import fs from "fs"
+import chalk from "chalk"
+import { isProduction } from "lib/environment"
+
+export const assetMiddleware = () => {
+  if (isProduction) {
```

Comment:
I'm a little confused about this middleware. Since non-development environments will upload assets to [and serve them from] S3, the application won't need to respond to requests for assets, right?

Reply:
> We'll also need to distinguish the staging/production targets.
AFAIK the `S3_BUCKET` env var is already set appropriately (different target buckets) on staging / production

Reply:
@joeyAghion All throughout our template code we use a function called `asset` to retrieve files to display. So as to avoid unnecessary refactoring throughout our volumous codebase I kept this implementation mapped 1:1 with the bucket assets implementation. That way when not in prod the function is a pass-through: 

```
// prod 
file => lookup(file)

// dev
file => file 
```

Does this answer your question?